runners/google-cloud-dataflow-java/worker/src/test/java/org/apache/beam/runners/dataflow/worker/fn/logging/BeamFnLoggingServiceTest.java

```diff
@@ -141,13 +144,16 @@ public void testMultipleClientsFailingIsHandledGracefullyByServer() throws Excep
                               .withOnError(waitForTermination::countDown)
                               .build());
               outboundObserver.onNext(createLogsWithIds(instructionId, -instructionId));
-              outboundObserver.onError(new RuntimeException("Client " + instructionId));
-              waitForTermination.await();
+              outboundObservers.add(outboundObserver);
               return null;
             });
       }
       ExecutorService executorService = Executors.newCachedThreadPool();
       executorService.invokeAll(tasks);
+      for (int i = 1; i <= 3; ++i) {
+        outboundObservers.take().onError(new RuntimeException("Client " + i));
```

Comment:
AFAICT this way of writing the test more clearly reads what it is supposed to do anyhow. This change LGTM.

Reply:
+1 of course this is the purpose of a countdown latch :-)

Reply:
Got it. Thank you.